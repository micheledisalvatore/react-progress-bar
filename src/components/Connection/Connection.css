.connection {
    width: calc(100% - 2.4em);
    height: 0.25em;
    background: #DBDBDB;
    position: absolute;
    bottom: 1.3em;
    left: calc(1.2em - 50%);
    z-index: 0;
}

.connection:after {
    content: '';
    background-color: #5131D7;
    height: 0.25em;
    left: 0;
    position: absolute;
    transition: width 500ms ease-in-out;
    width: 0;
    z-index: 2;
}

.connection--is-active:after {
    width: 100%;
    transition: width 500ms ease-in-out;
}