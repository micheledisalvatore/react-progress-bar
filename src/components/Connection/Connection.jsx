import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import './Connection.css';

export const Connection = ({ isActive }) => {
    const className = classNames('connection', {
        'connection--is-active': isActive,
    });

  return <div className={className} />;
};

Connection.propTypes = {
  isActive: PropTypes.bool,
};

Connection.defaultProps = {
  isActive: false,
};