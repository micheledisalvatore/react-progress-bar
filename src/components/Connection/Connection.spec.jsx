import React from 'react';
import { shallow } from 'enzyme';

import { Connection } from './Connection';

describe('Given a Connection component', () => {
  describe('when it is rendered', () => {
    const component = <Connection />;
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(component);
    });

    it('should match the snapshot', () => {
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('should render with NOT the active class', () => {
      expect(wrapper).not.toHaveClassName('connection--is-active');
    });

    describe('and it receives isActive prop', () => {
      beforeEach(() => {
        wrapper.setProps({
          isActive: true,
        });
      });

      it('should render with the active class', () => {
        expect(wrapper).toHaveClassName('connection--is-active');
      });
    });
  });
});