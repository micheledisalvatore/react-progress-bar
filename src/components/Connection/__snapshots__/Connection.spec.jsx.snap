// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Connection component when it is rendered should match the snapshot 1`] = `
<div
  className="connection"
/>
`;
